graphql/database/users/logSearch.js

```diff
@@ -82,7 +84,55 @@ const logSearch = async (userContext, userId, searchData = {}) => {
   } catch (e) {
     throw e
   }
-  return user
+  return {
+    user,
+    success: true,
+  }
+}
+
+const logSearchAnonUser = async (userContext, anonUserId, searchData) => {
+  try {
+    const source = getSource(searchData)
+    await UserSearchLogModel.create(userContext, {
+      userId: anonUserId,
+      timestamp: moment.utc().toISOString(),
+      ...(source && { source }),
+      ...(searchData.searchEngineId && {
+        searchEngine: searchData.searchEngineId,
+      }),
+      ...(searchData.causeId && { causeId: searchData.causeId }),
+      isAnonymous: true,
+      version: 1,
+    })
+  } catch (e) {
+    throw e
+  }
+  return {
+    success: true,
+  }
+}
+
+/**
+ * Log a user's search event and change related stats.
+ * @param {object} userContext - The user authorizer object.
+ * @param {string} userId - The user id.
+ * @param {string} anonUserId - The anonymous user id.
+ * @return {Promise<User>} A promise that resolves into a User instance.
+ */
+const logSearch = async (userContext, userId, anonUserId, searchData = {}) => {
+  if (userId && anonUserId) {
+    throw new Error('userId and anonUserId cannot be set at once.')
```

Comment:
Let's allow this case: prioritize using the userId if it exists, else use anonymous. (This is just one less error case to think about.)